resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientUtils.scala

```diff
@@ -51,6 +54,29 @@ private[spark] object KubernetesClientUtils extends Logging {
     propertiesWriter.toString
   }
 
+  object StringLengthOrdering extends Ordering[(String, String)] {
+    override def compare(x: (String, String), y: (String, String)): Int = {
+      // compare based on file length and break the tie with string comparison of keys.
+      (x._1.length + x._2.length).compare(y._1.length + y._2.length) * 10 +
+        x._1.compareTo(y._1)
+    }
+  }
+
+  def truncateToSize(seq: Seq[(String, String)], maxSize: Long): Map[String, String] = {
+    // First order the entries in order of their size.
+    // Skip entries if the resulting Map size exceeds maxSize.
+    val ordering: Ordering[(String, String)] = StringLengthOrdering
+    val sortedSet = SortedSet[(String, String)](seq : _*)(ordering)
+    var i: Int = 0
+    val map = mutable.HashMap[String, String]()
+    for (item <- sortedSet) {
+      i += item._1.length + item._2.length
+      if ( i < maxSize) {
```

Comment:
```scala
- if ( i < maxSize) {
+ if (i < maxSize) {
```

Reply:
And no big deal but if they're sorted by sum, you can stop after you hit something of maxSize

Reply:
P.S. I have evaluated the approach of trying to extend SortedSet, and some how make it size limited i.e. items cannot be added beyond a size is reached. But that was more complex in terms of code, so here I have chosen a piece of code that is more simpler to maintain and does not need an overhauling every time we upgrade scala major version.